### ISSUE TITLE
Initial install gets "TypeError: Cannot set properties of null (setting 'dev')"

### ISSUE THREAD

**Issue by VashJuan** [joaompneves/tsviz#53]
This tool looks super great, but install seemed to fail:
I've an Evergreen system (on VSCode, Win 10 Pro) having done `npm update -g`,
Trying `npm install -g tsviz-cli` 
returned:
npm ERR! Cannot set properties of null (setting 'dev')
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\[...]\npm-cache\_logs\2022-02-21T16_51_28_618Z-debug-0.log

I got the following error log - which doesn't give me enough to understand what is wrong or needed. Any ideas?!
(Am I supposed to install GraphViz first? You doc didn't seem to indicate that was necessary...)

0 verbose cli [
0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
0 verbose cli   'C:\\Users\\[...]\\npm\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'install',
0 verbose cli   '-g',
0 verbose cli   'tsviz-cli'
0 verbose cli ]
1 info using npm@8.5.1
2 info using node@v17.0.1
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 2ms
5 timing config:load:file:C:\Users\[...]\npm\node_modules\npm\npmrc Completed in 2ms
6 timing config:load:builtin Completed in 2ms
7 timing config:load:cli Completed in 2ms
8 timing config:load:env Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:C:\Users\[...]\.npmrc Completed in 1ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:C:\Users\[...]\npm\etc\npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:validate Completed in 0ms
15 timing config:load:credentials Completed in 1ms
16 timing config:load:setEnvs Completed in 1ms
17 timing config:load Completed in 10ms
18 timing npm:load:configload Completed in 10ms
19 timing npm:load:setTitle Completed in 0ms
20 timing config:load:flatten Completed in 2ms
21 timing npm:load:display Completed in 4ms
22 verbose logfile C:\Users\[...]\npm-cache\_logs\2022-02-21T16_51_28_618Z-debug-0.log
23 timing npm:load:logFile Completed in 4ms
24 timing npm:load:timers Completed in 1ms
25 timing npm:load:configScope Completed in 0ms
26 timing npm:load Completed in 19ms
27 timing arborist:ctor Completed in 1ms
28 silly logfile start cleaning logs, removing 1 files
29 timing idealTree:init Completed in 5ms
30 timing idealTree:userRequests Completed in 2ms
31 silly idealTree buildDeps
32 silly fetch manifest tsviz-cli@*
33 http fetch GET 200 https://registry.npmjs.org/tsviz-cli 326ms (cache miss)
34 silly placeDep ROOT tsviz-cli@2.0.0 OK for:  want: *
35 silly fetch manifest graphviz@>=0.0.8
36 silly fetch manifest tsviz@../tsviz
37 http fetch GET 200 https://registry.npmjs.org/graphviz 111ms (cache miss)
38 timing idealTree:#root Completed in 445ms
39 silly placeDep node_modules/tsviz-cli graphviz@0.0.9 OK for: tsviz-cli@2.0.0 want: >=0.0.8
40 silly placeDep node_modules/tsviz-cli tsviz@ OK for: tsviz-cli@2.0.0 want: ../tsviz
41 silly fetch manifest temp@~0.4.0
42 http fetch GET 200 https://registry.npmjs.org/temp 62ms (cache miss)
43 timing idealTree:node_modules/tsviz-cli Completed in 69ms
44 silly placeDep node_modules/tsviz-cli temp@0.4.0 OK for: graphviz@0.0.9 want: ~0.4.0
45 timing idealTree:node_modules/tsviz-cli/node_modules/graphviz Completed in 1ms
46 timing idealTree:node_modules/tsviz-cli/node_modules/temp Completed in 0ms
47 timing idealTree:node_modules/tsviz-cli/node_modules/tsviz Completed in 0ms
48 timing idealTree:node_modules/tsviz Completed in 0ms
49 timing idealTree:buildDeps Completed in 516ms
50 timing idealTree:fixDepFlags Completed in 1ms
51 timing idealTree Completed in 524ms
52 timing reify:loadTrees Completed in 527ms
53 timing reify:diffTrees Completed in 1ms
54 silly reify moves {}
55 timing reify:retireShallow Completed in 0ms
56 timing reify:createSparse Completed in 2ms
57 timing reify:loadBundles Completed in 0ms
58 silly audit bulk request { 'tsviz-cli': [ '2.0.0' ], graphviz: [ '0.0.9' ], temp: [ '0.4.0' ] }
59 silly tarball no local data for temp@https://registry.npmjs.org/temp/-/temp-0.4.0.tgz. Extracting by manifest.
60 timing reifyNode:node_modules/tsviz-cli/node_modules/tsviz Completed in 9ms
61 silly tarball no local data for graphviz@https://registry.npmjs.org/graphviz/-/graphviz-0.0.9.tgz. Extracting by manifest.
62 silly tarball no local data for tsviz-cli@https://registry.npmjs.org/tsviz-cli/-/tsviz-cli-2.0.0.tgz. Extracting by manifest.
63 http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 72ms
64 timing auditReport:getReport Completed in 74ms
65 silly audit report {}
66 timing auditReport:init Completed in 0ms
67 timing reify:audit Completed in 74ms
68 http fetch GET 200 https://registry.npmjs.org/graphviz/-/graphviz-0.0.9.tgz 169ms (cache miss)
69 timing reifyNode:node_modules/tsviz-cli/node_modules/graphviz Completed in 180ms
70 http fetch GET 200 https://registry.npmjs.org/temp/-/temp-0.4.0.tgz 213ms (cache miss)
71 timing reifyNode:node_modules/tsviz-cli/node_modules/temp Completed in 220ms
72 http fetch GET 200 https://registry.npmjs.org/tsviz-cli/-/tsviz-cli-2.0.0.tgz 254ms (cache miss)
73 timing reifyNode:node_modules/tsviz-cli Completed in 268ms
74 timing reify:unpack Completed in 268ms
75 timing reify:unretire Completed in 0ms
76 timing build:queue Completed in 0ms
77 timing build:deps Completed in 0ms
78 timing build:queue Completed in 0ms
79 timing build:links Completed in 0ms
80 timing build Completed in 1ms
81 timing reify:build Completed in 1ms
82 timing reify:trash Completed in 1ms
83 timing command:install Completed in 813ms
84 verbose stack TypeError: Cannot set properties of null (setting 'dev')
84 verbose stack     at calcDepFlagsStep (C:\Users\[...]\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\calc-dep-flags.js:34:21)
84 verbose stack     at visit (C:\Users\[...]\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\calc-dep-flags.js:12:20)
84 verbose stack     at visitNode (C:\Users\[...]\npm\node_modules\npm\node_modules\treeverse\lib\depth-descent.js:57:25)
84 verbose stack     at next (C:\Users\[...]\npm\node_modules\npm\node_modules\treeverse\lib\depth-descent.js:44:19)
84 verbose stack     at depth (C:\Users\[...]\npm\node_modules\npm\node_modules\treeverse\lib\depth-descent.js:82:10)
84 verbose stack     at depth (C:\Users\[...]\npm\node_modules\npm\node_modules\treeverse\lib\depth.js:27:12)
84 verbose stack     at calcDepFlags (C:\Users\[...]\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\calc-dep-flags.js:10:15)
84 verbose stack     at Arborist.[copyIdealToActual] (C:\Users\[...]\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\reify.js:1495:7)
84 verbose stack     at Arborist.reify (C:\Users\[...]\npm\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\reify.js:156:35)
84 verbose stack     at async Install.exec (C:\Users\[...]\npm\node_modules\npm\lib\commands\install.js:157:5)
85 verbose cwd D:\Projects\RangerTrak\rangertrak
86 verbose Windows_NT 10.0.19044
87 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\[...]\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "tsviz-cli"
88 verbose node v17.0.1
89 verbose npm  v8.5.1
90 error Cannot set properties of null (setting 'dev')
91 verbose exit 1
92 timing npm Completed in 1089ms
93 verbose unfinished npm timer reify 1645462288892
94 verbose code 1
95 error A complete log of this run can be found in:
95 error     C:\Users\[...]\npm-cache\_logs\2022-02-21T16_51_28_618Z-debug-0.log

**Comment by VashJuan** [comment]
I installed latest GraphViz, had it update the PATH, restarted VSCode, retried `npm install -g tsviz-cli` but got the same error. Hmmm.

**Comment by VashJuan** [comment]
I've not been able to resolve this. Any ideas would be appreciated.

**Comment by VashJuan** [comment]
Oh, I was able to seemingly get a local, not global install to work:

And searching a bit more I see this might be a recent cli error?
https://github.com/npm/cli/issues/3901
https://github.com/npm/cli/issues/4064
https://github.com/npm/cli/issues/3711

Hmmm. 
Tried `npm ci`

Now I'm getting:
tsviz-cli: The term 'tsviz-cli' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

Off to check package.json & node_modules...
